backend/routes/posts.js

```diff
@@ -32,9 +32,9 @@ router.get('/latest', cacheHandler(REDIS_KEYS.LATEST_POSTS), getLatestPostsHandl
 router.get('/:id', getPostByIdHandler);
 
 // Update a post by ID
-router.patch('/:id', updatePostHandler);
+router.patch('/:id', authenticationHandler, adminHandler, updatePostHandler);
 
```

Comment:
What if the post owner wants to update or delete the post, which we are even considering as the scope of the task, in that case, ig we'll fail here, can we take post author also into consideration here.

Reply:
The post model schema does not contain userid as the author of the post. I have implemented the author middleware but it will not be valid until we change the post-model schema.